src/state_machine.rs

```diff
@@ -544,117 +525,23 @@ impl StateMachine {
-impl StateMachine {
-    /// Query for a result, or yield: Err(NothingAvailable), Err(Disconnected) or Err(Terminated).
-    pub fn try_step(&mut self, outbox: &mut dyn EventBox) -> Result<(), mpmc::TryRecvError> {
-        if self.is_running {
-            match self.network_rx.try_recv() {
-                Ok(event) => {
-                    self.handle_network_event(event, outbox);
-                    return Ok(());
-                }
-                Err(mpmc::TryRecvError::Empty) => (),
-                Err(error) => return Err(error),
-            }
-
-            let action = self.action_rx.try_recv()?;
-            self.handle_action(action, outbox);
-            Ok(())
-        } else {
-            Err(mpmc::TryRecvError::Disconnected)
-        }
+impl Display for StateMachine {
+    fn fmt(&self, formatter: &mut Formatter) -> fmt::Result {
+        write!(formatter, "{:?}", self.state)
     }
 }
 
-#[cfg(feature = "mock_base")]
-impl StateMachine {
-    // Handle an event from the list and send any events produced for higher layers.
-    fn handle_event_from_list(&mut self, outbox: &mut dyn EventBox) {
-        assert!(!self.events.is_empty());
-        let event = self.events.remove(0);
-        match event {
-            EventType::Action(action) => self.handle_action(*action, outbox),
-            EventType::NetworkEvent(event) => self.handle_network_event(event, outbox),
-        };
-    }
-
-    /// Query for a result, or yield: Err(NothingAvailable), Err(Disconnected).
-    pub fn try_step(&mut self, outbox: &mut dyn EventBox) -> Result<(), mpmc::TryRecvError> {
-        use itertools::Itertools;
-        use rand::seq::SliceRandom;
-        use std::iter;
-
-        if !self.is_running {
-            return Err(mpmc::TryRecvError::Disconnected);
-        }
-
-        let mut events = Vec::new();
-        let mut timed_out_events = Vec::new();
-        let mut received = true;
-
-        // Populate action_rx timeouts
-        self.state.process_timers();
-
-        while received {
-            received = false;
-
-            match self.network_rx.try_recv() {
-                Ok(event) => {
-                    received = true;
-                    events.push(EventType::NetworkEvent(event));
-                }
-                Err(mpmc::TryRecvError::Empty) => (),
-                Err(mpmc::TryRecvError::Disconnected) => {
-                    self.apply_transition(Transition::Terminate, outbox);
-                    return Ok(());
-                }
-            }
-
-            if let Ok(action) = self.action_rx.try_recv() {
-                received = true;
-                let queue = match &action {
-                    Action::HandleTimeout(_) => &mut timed_out_events,
-                    _ => &mut events,
-                };
-                queue.push(EventType::Action(Box::new(action)));
-            }
-        }
-
-        // Interleave timer events with routing or network events.
```

Comment:
Note that this interleaving no longer happen, I think, so it may reduce the test coverage a bit.

Reply:
Yeah I know. Not sure we really need it. Also, [the crossbeam-channel docs](https://docs.rs/crossbeam-channel/0.4.0/crossbeam_channel/struct.Select.html) say this:

> Select allows you to define a set of channel operations, wait until any one of them becomes ready, and finally execute it. If multiple operations are ready at the same time, a **random one among them is selected**.

So maybe we get that feature for free now?

Reply:
Hm, but that can break determinism :thinking:

Reply:
Yes, I think you are correct, we do get it somewhat for free. the select is "random" I think in a deterministic simple way, but that is likely enough for our purpose.

Reply:
OK, so you don't think it breaks test repeatability then?

Reply:
> m, but that can break determinism
No, no worries here is it a reproducible random if I remember correctly when I looked it up

Reply:
````
pub fn shuffle<T>(v: &mut [T]) {
    let len = v.len();
    if len <= 1 {
        return;
    }

    thread_local! {
        static RNG: Cell<Wrapping<u32>> = Cell::new(Wrapping(1406868647));
    }
```
Yes, I think we are good.

Reply:
Yeah, just looking at it. It uses hard-coded seed so we should be good.